ivadomed/loader/generalized_loader_configuration.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import copy
+import typing
+from typing import List, Tuple
+
+from ivadomed.keywords import ROIParamsKW
+from ivadomed.transforms import Compose, prepare_transforms
+
+if typing.TYPE_CHECKING:
+    from ivadomed.loader.patch_filter import PatchFilter
+    from ivadomed.loader.slice_filter import SliceFilter
+
+
+class GeneralizedLoaderConfiguration:
+    """
+    A generalized class to store ALL key data requirement for the loading purpose.
+    It is a ONE-STOP configuration data class to enable subsequent downstream analyses.
+    Used to significantly simply the constructor parameters
+    """
+
+    def __init__(
+            self,
+            # Mandatory Parameters
+            model_params: dict,
+            # Optional Parameters
+            object_detection_params: dict = None,
+            transform: list = None,
+            nibabel_cache: bool = True,
+            disk_cache: bool = True,
+            metadata_choice: bool = False,
+            multichannel: bool = False,
+            soft_gt: bool = False,
+            is_input_dropout: bool = False,
+            slice_axis: int = 2,
+            slice_filter_fn: callable = None,
+            patch_filter_fn: callable = None,
+            roi_params: dict = None,
+            task: str = "segmentation",
+            length: Tuple[int, int, int] = (64, 64, 64),
+            stride: Tuple[int, int, int] = (0, 0, 0),
```

Comment:
In particular, I'm not sure how we should deal with `length` and `stride`.
Note that we have `length/stride_2D` that are _optional_ in `default_model` and of dimension 2, whereas we also have `length/stride_3D` that are _required_ in `Modified3DUNet` and of dimension 3.

Again, I don't understand how `length` and `stride` from here will be use down the line. I also wonder if other parameters have different _default_ values between 2D and 3D and if that will be an issue or not.

Reply:
Good point. I think it would be worth while to probably change these to NONE for now and expect upstream overwrite. 

The changes here are suppose to be a shim layer that is mirroring underlying values. Whatever existing config will overwrite them as before so none of it should in theory change but this length/stride dimensionality issue is a good example of 2D vs 3D difference.

Reply:
> Good point. I think it would be worth while to probably change these to NONE for now and expect upstream overwrite.

Yes, especially because the default is not the same for 2D and 3D as explained above.